src/MultiFormats.Tests/MultiHashTest.cs

```diff
@@ -0,0 +1,558 @@
+            Assert.IsFalse(c == b);
+            Assert.IsFalse(b == c);
+
+            Assert.IsFalse(c != d);
+            Assert.IsTrue(c != b);
+            Assert.IsTrue(b != c);
+
+#pragma warning disable 1718
+            Assert.IsTrue(a0 == a0);
+            Assert.IsTrue(a0 == a1);
+            Assert.IsFalse(a0 == b);
+
+#pragma warning disable 1718
+            Assert.IsFalse(a0 != a0);
+            Assert.IsFalse(a0 != a1);
+            Assert.IsTrue(a0 != b);
+
+            Assert.IsTrue(a0.Equals(a0));
+            Assert.IsTrue(a0.Equals(a1));
+            Assert.IsFalse(a0.Equals(b));
+
+            Assert.AreEqual(a0, a0);
+            Assert.AreEqual(a0, a1);
+            Assert.AreNotEqual(a0, b);
+
+            Assert.AreEqual(a0, a0);
+            Assert.AreEqual(a0, a1);
+            Assert.AreNotEqual(a0, b);
+
+            Assert.AreEqual(a0.GetHashCode(), a0.GetHashCode());
+            Assert.AreEqual(a0.GetHashCode(), a1.GetHashCode());
+            Assert.AreNotEqual(a0.GetHashCode(), b.GetHashCode());
+        }
+
+        [TestMethod]
+        public void Varint_Hash_Code_and_Length()
+        {
+            var concise = "1220f8c3bf62a9aa3e6fc1619c250e48abe7519373d3edf41be62eb5dc45199af2ef"
+               .ToHexBuffer();
+            var mh = new MultiHash(new MemoryStream(concise, false));
+            Assert.AreEqual("sha2-256", mh.Algorithm.Name);
+            Assert.AreEqual(0x12, mh.Algorithm.Code);
+            Assert.AreEqual(0x20, mh.Algorithm.DigestSize);
+
+            var longer = "9200a000f8c3bf62a9aa3e6fc1619c250e48abe7519373d3edf41be62eb5dc45199af2ef"
+               .ToHexBuffer();
+            mh = new MultiHash(new MemoryStream(longer, false));
+            Assert.AreEqual("sha2-256", mh.Algorithm.Name);
+            Assert.AreEqual(0x12, mh.Algorithm.Code);
+            Assert.AreEqual(0x20, mh.Algorithm.DigestSize);
+        }
+
+        [TestMethod]
+        public void Compute_Hash_All_Algorithms()
+        {
+            foreach (var alg in HashingAlgorithm.All)
+                try
+                {
+                    var mh = MultiHash.ComputeHash(new byte[0], alg.Name);
+                    Assert.IsNotNull(mh, alg.Name);
+                    Assert.AreEqual(alg.Code, mh.Algorithm.Code, alg.Name);
+                    Assert.AreEqual(alg.Name, mh.Algorithm.Name, alg.Name);
+                    Assert.AreEqual(alg.DigestSize, mh.Algorithm.DigestSize, alg.Name);
+                    Assert.AreEqual(alg.DigestSize, mh.Digest.Length, alg.Name);
+                }
+                catch (NotImplementedException)
+                {
+                    // If NYI then can't test it.
+                }
+        }
+
+        [TestMethod]
+        public void Example()
+        {
+            var hello = Encoding.UTF8.GetBytes("Hello world");
+            var mh = MultiHash.ComputeHash(hello);
+            Console.WriteLine($"| hash code | 0x{mh.Algorithm.Code.ToString("x")} |");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)